tests/integration/conftest.py

```diff
@@ -794,66 +789,129 @@ async def traefik_application_fixture(model: Model):
 
 @pytest_asyncio.fixture(scope="module", name="oathkeeper_related")
 async def oathkeeper_application_related_fixture(
-    ops_test: OpsTest, application: Application, client: Client, ext_idp_service: str
+    application: Application, client: Client, ext_idp_service: str, model: Model
 ):
     """The application related to Jenkins via auth_proxy v0 relation."""
-    oathkeeper = await application.model.deploy("oathkeeper", channel="edge", trust=True)
-    # Using a patched local bundle from identity team here as a temporary workaround for
-    # https://github.com/canonical/traefik-k8s-operator/issues/322 and
-    # https://github.com/juju/python-libjuju/issues/1042
-    await ops_test.run(
-        "juju", "deploy", "./tests/integration/files/identity-bundle-edge-patched.yaml", "--trust"
-    )
-    await application.model.applications["kratos-external-idp-integrator"].set_config(
-        {
+    oathkeeper = await model.deploy("oathkeeper", channel="edge", trust=True)
+    kratos_app = await model.deploy(
+        "kratos",
+        channel="0.4/edge",
+        # Needed per https://github.com/canonical/oathkeeper-operator/issues/49
+        config={"dev": "True"},
+        trust=True,
+    )
+    kratos_external_idp_integrator_app = await model.deploy(
+        "kratos-external-idp-integrator",
+        channel="latest/edge",
+        config={
             "client_id": "client_id",
             "client_secret": "client_secret",
             "provider": "generic",
             "issuer_url": ext_idp_service,
             "scope": "profile email",
             "provider_id": "Dex",
-        }
+        },
     )
-
-    # See https://github.com/canonical/kratos-operator/issues/182
-    await application.model.wait_for_idle(
-        status="active",
-        apps=[application.name, oathkeeper.name] + IDENTITY_PLATFORM_APPS,
-        raise_on_error=False,
-        timeout=30 * 60,
-        idle_period=5,
+    hydra_app = await model.deploy("hydra", channel="edge", series="jammy", trust=True)
```

Comment:
Re-written due to race condition in IDP charms.
https://github.com/canonical/kratos-external-idp-integrator/issues/149